submissions/Roophee/a-tiny-js-world-oop/index.js

```diff
@@ -0,0 +1,68 @@
+"use strict";
+
+const delimiter = '; '
+
+class Inhabitant  {
+    constructor(name, gender, saying, legs=4, tail= 1 ){
+        this.name = name;
+        this.gender = gender;
+        this.species = this.constructor.name.toLowerCase();
+        this.saying = saying;
+        this.legs = legs;
+        this.tail = tail;
+        this.friends = [];
+    }
+    show() {
+        console.log(this.constructor)
+    }
+
+    addFriends(...arrayOfFriends) {
+        arrayOfFriends.forEach(({name}) => this.friends.push(name));
+    }
+
+    description(){
+        return Object.entries(this).map(property => `${property[0]}: ${property[1]}`).join(delimiter);
+    }
```

Comment:
Let's take a step back.
The following from the [earlier requirements](https://github.com/kottans/frontend/blob/master/tasks/js-pre-oop.md#additional-requirements) is valid at all times.

![image](https://user-images.githubusercontent.com/8686631/103414346-3f9eba80-4b86-11eb-8ab3-31b267a78c4a.png)

Reply:
Can I use:
for ( let property in obj) {...} ?

Reply:
Or you mean I should create athe array of properties wich is nessary to output and I can use "map" function to enumerate properties with call obj.property?

Reply:
Yes. But rather `obj[propertyName]`
https://github.com/kottans/frontend-2021-homeworks/pull/377#discussion_r550455922

Reply:
Done.